dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -70,66 +72,162 @@ class Identity[A] extends Transformer[A, A] {
   }
 }
 
+
 object SampleToBatch {
   def apply[T: ClassTag]
-  (batchSize: Int)
+  (batchSize : Int,
+   padFeatureValue : Option[Tensor[T]] = None,
+   padLabelValue : Option[T] = None,
+   fixLength: Option[Int] = None)
   (implicit ev: TensorNumeric[T]): SampleToBatch[T]
-  = new SampleToBatch[T](batchSize)
+  = new SampleToBatch[T](batchSize, padFeatureValue, padLabelValue, fixLength)
 }
 
+/**
+ * convert sample to MiniBatch, and you can choose padding feature or label
+ * to same length in a batch
+ * @param totalBatch batchsize
+ * @param padFeatureValue  padding feature value if do padding
+ * @param padLabelValue padding label value if do padding
+ * @param fixLength padding fix length, default the max first dimension length of feature
+ *                  or label in a batch
+ */
 class SampleToBatch[T: ClassTag]
-  (totalBatch: Int)
-  (implicit ev: TensorNumeric[T])
+(totalBatch : Int,
+ padFeatureValue : Option[Tensor[T]] = None,
+ padLabelValue : Option[T] = None,
+ fixLength: Option[Int] = None)
+(implicit ev: TensorNumeric[T])
   extends Transformer[Sample[T], MiniBatch[T]] {
 
+  private def paddingTensor(data: Array[T], padValue: Tensor[T], start: Int, end: Int): Unit = {
+    var offset = start
+    val padArr = padValue.storage().array()
+    val padOffset = padValue.storageOffset() - 1
+    while (offset < end) {
+      val length = math.min(end - offset, padArr.length)
+      System.arraycopy(padArr, padOffset, data, offset, length)
+      offset += length
+    }
+  }
+
+  private def paddingValue(data: Array[T], padValue: T, start: Int, end: Int): Unit = {
+    ev.getType() match {
+      case DoubleType =>
+        util.Arrays.fill(data.asInstanceOf[Array[Double]], start, end, ev.toType[Double](padValue))
+      case FloatType =>
+        util.Arrays.fill(data.asInstanceOf[Array[Float]], start, end, ev.toType[Float](padValue))
+      case _ => throw new UnsupportedOperationException(
+        "SampleToBatchPadding: Only Float/Double supported")
+    }
+  }
+
+  private def getSize(padLength: Int, batchLength: Int, tensorSize: Array[Int]): Int = {
```

Comment:
The method name is confusing - it returns element# for the tensor, and then update tensorSize with batch size. Chane it to something meaningful, and add more comments.